### PR TITLE
Fix use of undefined channel when skipping binners

### PR DESCRIPTION
**Pull request by skrakau**
Fix use of undefined channel (`METABAT2_METABAT2.out`, `MAXBIN2.out`) in case of `--skip_metabat2` or `--skip_maxbin2` and restructured code.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/binning.nf [hunk 1]
```diff
@@ -159,11 +158,8 @@ workflow BINNING {
     ch_versions = ch_versions.mix(MAG_DEPTHS_SUMMARY.out.versions)
 
     // Group final binned contigs per sample for final output
-    ch_binning_results_gunzipped_final = ch_binning_results_gunzipped
-        .groupTuple(by: 0)
-
-    ch_binning_results_gzipped_final = METABAT2_METABAT2.out.fasta.mix(MAXBIN2.out.binned_fastas)
-        .groupTuple(by: 0)
+    ch_binning_results_gunzipped_final = ch_binning_results_gunzipped.groupTuple(by: 0)
+    ch_binning_results_gzipped_final   = ch_binning_results_gzipped_final.groupTuple(by: 0)
```
Comment (d4straub) [inline subworkflows/local/binning.nf:162]: Hm input & output same name might be error prone??

**Review by d4straub — APPROVED** [review]
Looks fine to me, only comment see above.

File: subworkflows/local/binning.nf (hunk 1 above)
Comment (skrakau) [reply subworkflows/local/binning.nf:162]: Good question... it's done for software versions all the time 
```
ch_versions = ch_versions.mix(<MY_PROCESS>.out.versions.first())
```
so I assumed it's ok, but might be a different case

Comment (d4straub) [reply subworkflows/local/binning.nf:162]: Hmyes, `groupTuple` and `mix` are quite different. I am just not sure.

**Comment by skrakau** [comment]
Thanks @d4straub !

**Pull request merged**